core/dispatcher/src/main/scala/whisk/core/invoker/Invoker.scala

```diff
@@ -332,27 +337,49 @@ class Invoker(
      *
      * Note: Updates the container's log cursor to indicate consumption of log.
      */
-    private def getContainerLogs(con: WhiskContainer, tries: Int = LogRetryCount)(implicit transid: TransactionId): JsArray = {
+    private def getContainerLogs(con: WhiskContainer, limits: ActionLimits, tries: Int = LogRetryCount)(implicit transid: TransactionId): JsArray = {
         val size = con.getLogSize(runningInContainer)
+        val maxSize = limits.logs.map { _.megabytes }.getOrElse(Int.MaxValue)
         val advanced = size != con.lastLogSize
         if (tries <= 0 || advanced) {
             val rawLogBytes = con.getDockerLogContent(con.lastLogSize, size, runningInContainer)
             val rawLog = new String(rawLogBytes, "UTF-8")
+
             val isNodeJs = con.image == nodejsImageName || con.image == nodejs6ImageName
-            var isSwift = con.image == swiftImageName || con.image == swift3ImageName
-            val (complete, logArray) = processJsonDriverLogContents(rawLog, isNodeJs || isSwift)
+            val isSwift = con.image == swiftImageName || con.image == swift3ImageName
+            val (complete, logArray) = processJsonDriverLogContents(rawLog, isNodeJs || isSwift, maxSize megabytes)
+
             if (tries > 0 && !complete) {
                 info(this, s"log cursor advanced but missing sentinel, trying $tries more times")
                 Thread.sleep(LogRetry)
-                getContainerLogs(con, tries - 1)
+                getContainerLogs(con, limits, tries - 1)
```

Comment:
should this pass LogLimit instead - since the other limits are not relevant to getContainerLogs.